SAFARI/BCAGPA.js

```diff
@@ -0,0 +1,164 @@
+        d["W"] = 2;
+        d["R"] = 3;
+        d["F"] = 4;
+
+        var mod = new Array(5);
+        for (var i = 0; i < 5; i++) {
+            mod[i] = new Array(3);
+            for (var j = 0; j < 3; j++) {
+                mod[i][j] = 0;
+            }
+        }
+        var t = str.split(" ");
+
+        var e = Math.min(str.indexOf("-"), str.indexOf("("));
+        var startMod = parseInt(str.substring(0, e));
+
+        for (var i = 0; i < t.length; i++) {
+            var b2 = t[i].indexOf("(");
+            var b1 = t[i].substring(0, b2).indexOf("-");
+            var b3 = t[i].indexOf(")");
+
+            var s, e;
+            if (b1 == -1) {
+                s = parseInt(t[i].substring(0, b2));
+                e = s;
+            } else {
+                s = parseInt(t[i].substring(0, b1));
+                e = parseInt(t[i].substring(b1 + 1, b2));
+            }
+
+            var parts = t[i].substring(b2 + 1, b3).split(",");
+
+            for (var j = 0; j < parts.length; j++) {
+                var bb = parts[j].indexOf("-");
+                var sd = d[parts[j][0]],
+                    ed;
+                if (bb == -1) ed = sd;
+                else ed = d[parts[j][2]];
+
+                for (var day = sd; day <= ed; day++) {
+                    for (var k = s; k <= e; k++) {
+                        mod[day][k - startMod] = 1;
+                    }
+                }
+            }
+        }
+        for (var i = 0; i < 5; i++) {
+            for (var j = 0; j < 3; j++) {
+                mods += mod[i][j];
+            }
+        }
+        return mods;
+    }
+
+    function classData() {
+        src = document.documentElement.outerHTML;
+        var classes = document.querySelectorAll('tr[id^="ccid"]');
+
+        var data = [];
+
+        for (var i = 0; i < classes.length; i++) {
+            var name = classes.item(i).querySelector('td[class="table-element-text-align-start"]');
+            name = name.textContent;
+            if (name[0] == "~") continue; // Not included in GPA
+
+            var info = classes.item(i).querySelectorAll("td");
+            var grades = ["N", "N", "N", "N", 0];
+
+            var hours = info.item(0).textContent;
+            grades[4] = modsPerWeek(hours);
+
+            for (var j = 0; j < info.length; j++) {
+                var elta = info.item(j).querySelector("a");
+                if (elta == null) continue;
+                let href = elta.href.substring(elta.href.length - 14, elta.href.length - 12);
+                grades[["T1", "T2", "T3", "Y1"].indexOf(href)] = extractLetterGrade(elta.textContent);
+            }
+            data.push(grades);
+        }
+        return data;
+    }
+
+    var data = classData();
+    var pts = {
+        A: 4.0,
+        P: 4.0,
+        "A-": 3.8,
+        "B+": 3.3,
+        B: 3.0,
+        "B-": 2.8,
+        "C+": 2.3,
+        C: 2.0,
+        "C-": 1.8,
+        "D+": 1.3,
+        D: 1.1,
+        F: 0.0
```

Comment:
Some are missing quotes

Reply:
I am using the Prettier formatting guide for this. Javascript syntax doesn't require you to have quotes around keys (this is a js object not a json) so it only put the quotes around the keys that have symbols. I can look into having prettier put quotes around all keys in JS objects if you'd like

Reply:
If possible can you try and have it have quotes around all of them? It looks strange that only some of them do and others don't. If it's not too much work don't worry about it

Reply:
manually set them all to have quotes for now, prettier still forces them back to no quotes so we'll just have to not format that block with prettier